rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful.
+ * \param[in] allocator The allocator to use for memory allocation.
+ * \return RCL_LOGGING_RET_OK if successful, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
+RCL_LOGGING_INTERFACE_PUBLIC
+inline
+rcl_logging_ret_t
+rcl_logging_get_logging_directory(const char ** directory, rcutils_allocator_t allocator)
+{
+  const char * log_dir_env;
```

Comment:
It would be nice to return `RCL_LOGGING_RET_INVALID_ARG` if either:
- `directory` is `NULL` (to avoid segfaults).
- `*directory` is not NULL (to avoid leaks).

Reply:
done as part of 3d5bfb3d86138d241475940502b52179f9910628

I wrote quite literal error messages/doc, so let me know if that's okay.